src/wallet/wallet.cpp

```diff
@@ -2602,94 +2602,111 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain* chain, const std::st
     }
 
     if (gArgs.IsArgSet("-mintxfee")) {
-        CAmount n = 0;
-        if (!ParseMoney(gArgs.GetArg("-mintxfee", ""), n) || 0 == n) {
-            error = AmountErrMsg("mintxfee", gArgs.GetArg("-mintxfee", ""));
-            return nullptr;
-        }
-        if (n > HIGH_TX_FEE_PER_KB) {
+        if (std::optional<CAmount> min_tx_fee = ParseMoney(gArgs.GetArg("-mintxfee", ""))) {
```

Comment:
I think you could achieve the same, with a smaller diff and less code duplication as follows:

```diff
--- a/src/wallet/wallet.cpp
+++ b/src/wallet/wallet.cpp
@@ -2602,22 +2602,16 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain* chain, const std::st
     }
 
     if (gArgs.IsArgSet("-mintxfee")) {
-        if (std::optional<CAmount> min_tx_fee = ParseMoney(gArgs.GetArg("-mintxfee", ""))) {
-            if (min_tx_fee.value() == 0) {
-                error = AmountErrMsg("mintxfee", gArgs.GetArg("-mintxfee", ""));
-                return nullptr;
-            }
-
-            if (min_tx_fee.value() > HIGH_TX_FEE_PER_KB) {
-            warnings.push_back(AmountHighWarn("-mintxfee") + Untranslated(" ") +
-                               _("This is the minimum transaction fee you pay on every transaction."));
-            }
-
-            walletInstance->m_min_fee = CFeeRate(min_tx_fee.value());
-        } else {
+        auto min_tx_fee = ParseMoney(gArgs.GetArg("-mintxfee", ""));
+        if (!min_tx_fee || min_tx_fee.value() == 0) {
             error = AmountErrMsg("mintxfee", gArgs.GetArg("-mintxfee", ""));
             return nullptr;
+        } else if (min_tx_fee.value() > HIGH_TX_FEE_PER_KB) {
+            warnings.push_back(AmountHighWarn("-mintxfee") + Untranslated(" ") +
+                               _("This is the minimum transaction fee you pay on every transaction."));
         }
+
+        walletInstance->m_min_fee = CFeeRate(min_tx_fee.value());
     }
 
     if (gArgs.IsArgSet("-maxapsfee")) {
```

Reply:
Suggested the same in https://github.com/bitcoin/bitcoin/pull/22220#discussion_r681491525 ;)

Reply:
Updated to take your suggestions.